homeworks/homework8/task2.py

```diff
@@ -0,0 +1,73 @@
+"""Write a wrapper class TableData for database table, that
+ when initialized with database name and table acts as
+ collection object (implements Collection protocol). Assume
+ all data has unique values in 'name' column. So, if
+
+presidents = TableData(database_name='example.sqlite', table_name='presidents')
+
+ then:
+- len(presidents) will give current amount of rows in
+  presidents table in database
+- presidents['Yeltsin'] should return single data row
+  for president with name Yeltsin
+- 'Yeltsin' in presidents should return if president
+  with same name exists in table
+- object implements iteration protocol. i.e. you could use it in for loops::
+
+for president in presidents:
+    print(president['name'])
+
+- all above mentioned calls should reflect most recent data.
+ If data in table changed after you created collection instance,
+ your calls should return updated data.
+
+Avoid reading entire table into memory. When iterating through records,
+start reading the first record, then go to the next one, until records
+are exhausted. When writing tests, it's not always neccessary to mock
+database calls completely. Use supplied example.sqlite file as database
+fixture file."""
+import sqlite3
+
+
+class TableData:
+
+    def __init__(self, database_name: str, table_name: str):
+        self.database_name = database_name
+        self.table_name = table_name
+
+    def __len__(self):
+        with sqlite3.connect(f'{self.database_name}') as conn:
+            conn.row_factory = sqlite3.Row
+            cursor = conn.cursor()
+            cursor.execute(f'SELECT COUNT(*) from {self.table_name}')
+        return cursor.fetchone()[0]
+
+    def __iter__(self):
+        with sqlite3.connect(f'{self.database_name}') as conn:
+            conn.row_factory = sqlite3.Row
+            cursor = conn.cursor()
+            cursor.execute(f'SELECT * from {self.table_name}')
+            self.tabledata = cursor.fetchall()
+            self._cursor = 0
+        return self
+
+    def __next__(self):
+        if self._cursor < len(self.tabledata):
+            row = self.tabledata[self._cursor]
+            print(row)
+            result = dict(zip(row.keys(), row))
+            self._cursor += 1
+            return result
+        raise StopIteration
```

Comment:
u should iterating and return line by line. u dont need to collect counter (_cursor)